tests/python/gaia-ui-tests/gaiatest/apps/contacts/regions/contact_form.py

```diff
@@ -161,7 +161,7 @@ def tap_update(self, return_details=True):
             return ContactDetails(self.marionette)
         else:
             # else we drop back to the underlying app
-            Wait(self.marionette).until(lambda m: self.apps.displayed_app.name != self.name)
+            Wait(self.marionette).until(lambda m: self.apps.displayed_app.manifest_url != self.manifest_url)
```

Comment:
Let's factorize this wait too, in `Base`, exactly the same way as we did in `wait_to_be_displayed()`